cmd/collector/app/builder/span_handler_builder.go

```diff
@@ -170,3 +179,56 @@ func fixConfiguration(config cascfg.Configuration) cascfg.Configuration {
 	config.ProtoVersion = 4
 	return config
 }
+
+type esSpanHandlerBuilder struct {
+	logger        *zap.Logger
+	client        es.Client
+	configuration escfg.Configuration
+}
+
+func newESBuilder(config *escfg.Configuration, logger *zap.Logger) *esSpanHandlerBuilder {
+	return &esSpanHandlerBuilder{
+		configuration: *config,
+		logger:        logger,
+	}
+}
+
+func (e *esSpanHandlerBuilder) BuildHandlers() (app.ZipkinSpansHandler, app.JaegerBatchesHandler, error) {
+	// TODO: keep these commented, add later when metrics are added to ES.
+	//hostname, _ := os.Hostname()
+	//hostMetrics := e.metricsFactory.Namespace(hostname, nil)
+
+	zSanitizer := zs.NewChainedSanitizer(
+		zs.NewSpanDurationSanitizer(e.logger),
+		zs.NewParentIDSanitizer(e.logger),
+	)
+
+	client, err := e.getClient()
+	if err != nil {
+		return nil, nil, err
+	}
+	spanStore := esSpanstore.NewSpanWriter(client, e.logger)
+
+	spanProcessor := app.NewSpanProcessor(
+		spanStore,
+		//app.Options.ServiceMetrics(e.metricsFactory),
+		//app.Options.HostMetrics(hostMetrics),
+		app.Options.Logger(e.logger),
+		app.Options.SpanFilter(defaultSpanFilter),
+		app.Options.NumWorkers(*NumWorkers),
+		app.Options.QueueSize(*QueueSize),
+	)
+
+	return app.NewZipkinSpanHandler(e.logger, spanProcessor, zSanitizer),
+		app.NewJaegerSpanHandler(e.logger, spanProcessor),
+		nil
```

Comment:
nope, sorry, this won't do. You've hit the rule of 3. Let's refactor this properly to avoid all the code dups (as a prerequisite PR).